src/System/IO/BlockIO/URingFFI.hsc

```diff
@@ -26,6 +29,136 @@ foreign import capi unsafe "liburing.h io_uring_queue_init"
 foreign import capi unsafe "liburing.h io_uring_queue_exit"
   io_uring_queue_exit :: Ptr URing -> IO ()
 
+foreign import capi safe "liburing.h io_uring_queue_init_params"
+  io_uring_queue_init_params :: CUInt -> Ptr URing -> Ptr URingParams -> IO CInt
+
+foreign import capi unsafe "liburing.h value IORING_SETUP_CQSIZE" iORING_SETUP_CQSIZE :: CUInt
+
+data {-# CTYPE "liburing.h" "struct io_uring_params" #-}
+     URingParams = URingParams {
+                      sq_entries :: !CUInt,
+                      cq_entries :: !CUInt,
+                      flags :: !CUInt,
+                      sq_thread_cpu :: !CUInt,
+                      sq_thread_idle :: !CUInt,
+                      features :: !CUInt,
+                      wq_fd :: !CUInt,
+                      resv1 :: !CUInt,
+                      resv2 :: !CUInt,
+                      resv3 :: !CUInt,
+                      sq_off :: !SQRingOffsets,
+                      cq_off :: !CQRingOffsets
```

Comment:
I think we can omit most of these, so long as we fill the whole struct to 0s when we `poke` it. That would save code and avoid conditional compilation for new fields.

I'll do that as a follow-up patch.